rpc/lib/server/handlers.go

```diff
@@ -349,80 +349,89 @@ const (
 type wsConnection struct {
 	cmn.BaseService
 
-	remoteAddr  string
-	baseConn    *websocket.Conn
-	writeChan   chan types.RPCResponse
-	readTimeout *time.Timer
-	pingTicker  *time.Ticker
+	remoteAddr string
+	baseConn   *websocket.Conn
+	writeChan  chan types.RPCResponse
 
 	funcMap map[string]*RPCFunc
 	evsw    events.EventSwitch
+
+	// write channel capacity
+	writeChanCapacity int
+
+	// each write times out after this.
+	writeWait time.Duration
+
+	// Connection times out if we haven't received *anything* in this long, not even pings.
+	readWait time.Duration
+
+	// Send pings to server with this period. Must be less than readWait, but greater than zero.
+	pingPeriod time.Duration
 }
 
-// new websocket connection wrapper
-func NewWSConnection(baseConn *websocket.Conn, funcMap map[string]*RPCFunc, evsw events.EventSwitch) *wsConnection {
+// NewWSConnection wraps websocket.Conn. See the commentary on the
+// func(*wsConnection) functions for a detailed description of how to configure
+// ping period and pong wait time.
+func NewWSConnection(baseConn *websocket.Conn, funcMap map[string]*RPCFunc, evsw events.EventSwitch, options ...func(*wsConnection)) *wsConnection {
 	wsc := &wsConnection{
-		remoteAddr: baseConn.RemoteAddr().String(),
-		baseConn:   baseConn,
-		writeChan:  make(chan types.RPCResponse, writeChanCapacity), // error when full.
-		funcMap:    funcMap,
-		evsw:       evsw,
+		remoteAddr:        baseConn.RemoteAddr().String(),
+		baseConn:          baseConn,
+		funcMap:           funcMap,
+		evsw:              evsw,
+		writeWait:         defaultWSWriteWait,
+		writeChanCapacity: defaultWSWriteChanCapacity,
+		readWait:          defaultWSReadWait,
+		pingPeriod:        defaultWSPingPeriod,
+	}
+	for _, option := range options {
+		option(wsc)
 	}
 	wsc.BaseService = *cmn.NewBaseService(nil, "wsConnection", wsc)
 	return wsc
 }
 
+func WriteWait(writeWait time.Duration) func(*wsConnection) {
```

Comment:
Yes in deed, I am familiar with them and I like them :) https://github.com/odeke-em/mapbox/blob/8bb614ae9e4db26d0a89d9fc1ea5bde442c72fca/options.go#L7-L21

However, my problem here is that these are exposed options ie exported options for potential concurrent misuse